cpptod.dd

```diff
@@ -829,6 +830,84 @@ void test()
 }
 ------
 
+<hr>$(COMMENT  -------------------------------------------- )
+
+$(H3 <a name="interfaces">Interfaces</a>)
+
+Interfaces are used to describe the behavior or capabilities of a class without
+commiting to a particular implementation of the class.
+
+$(H4 The C++ Way)
+
+  In C++, an interface is implemented with an `Abstract Class`. The abstract class
+	defines the class capabilities as pure virtual functions. A class which implements
+  the interface inherits from the abstract class and implements the pure virtual functions.
```

Comment:
The flow needs a bit work, how about:

"C++ interfaces are implemented by means of  $(I abstract classes). An abstract class defines interface methods as pure virtual functions. A class implementing the interface inherits the abstract class and implements its pure virtual functions."